easybuild/easyconfigs/s/Siesta/Siesta-5.2.2-foss-2023a.eb

```diff
@@ -25,12 +25,14 @@ sources = [
      'filename': 'elsi_interface_512532a42c5e.tar.gz',
      'extract_cmd': "tar xzf %s -C %(builddir)s/siesta-%(version)s/External/ELSI-project"},
 ]
-
-checksums = [
-    '4efc58fd5adb9b8f55234f47d11f89d268524b0d4c003d145d89d803da94a58c',  # siesta-5.2.2.tar.gz
-    '40651a9832eb93dec20a8360dd535262c261c34e13c41b6755fa6915c936b254',  # wannier90-3.1.0.tar.gz
-    None,  # elsi_interface_512532a42c5e.tar.gz
-]
+patches = ['Siesta-5.2.2-no_write_in_test.patch']
+
+checksums = [{
+  'siesta-5.2.2.tar.gz': '4efc58fd5adb9b8f55234f47d11f89d268524b0d4c003d145d89d803da94a58c',
+  'wannier90-3.1.0.tar.gz': '40651a9832eb93dec20a8360dd535262c261c34e13c41b6755fa6915c936b254',
+  'elsi_interface_512532a42c5e.tar.gz': None,
+  'Siesta-5.2.2-no_write_in_test.patch': '56c0f87dcb43144cfa328c800262634200dd960c79a85eb8c627f640cfc5686b',
+}]
```

Comment:
@ocaisa the tests failed with `'NoneType' object is not iterable"` here, I had had the same issue when using `None` in a dictionary of checksums, which is why I used the list format above, but it looks like with easybuild 5 this works as intended? I'll request a new test to check

Reply:
https://github.com/easybuilders/easybuild-easyconfigs/pull/22562/commits/eba634230b8579a5680aeeb0e53d3110a23bc382